pkg/storaged/dialogx.jsx

```diff
@@ -462,3 +464,128 @@ export const Skip = (className, options) => {
         }
     };
 };
+
+const StatelessSlider = ({ fraction, onChange }) => {
+    function start_dragging(event) {
+        let el = event.currentTarget;
+        let width = el.offsetWidth;
+        let left = el.offsetLeft;
+        while (el.offsetParent) {
+            el = el.offsetParent;
+            left += el.offsetLeft;
+        }
+
+        function drag(event) {
+            let f = (event.pageX - left) / width;
+            if (f < 0) f = 0;
+            if (f > 1) f = 1;
+            onChange(f);
+        }
+
+        function stop_dragging() {
+            document.removeEventListener("mousemove", drag);
+            document.removeEventListener("mouseup", stop_dragging);
+        }
+
+        document.addEventListener("mousemove", drag);
+        document.addEventListener("mouseup", stop_dragging);
+        drag(event);
+    }
+
+    if (fraction < 0) fraction = 0;
+    if (fraction > 1) fraction = 1;
+
+    return (
+        <div className="slider" onMouseDown={start_dragging}>
+            <div className="slider-bar" style={{ width: fraction * 100 + "%" }}>
```

Comment:
Does that work with `no-inline` C-S-P?

Reply:
Yes.  C-S-P probably only applies to parsing HTML, not constructing DOM nodes programmatically.